### ISSUE TITLE
Docstring from __init__ method is placed just after class method docstring in Circuit Library and looks strange.

### ISSUE THREAD

**Issue by Guillermo-Mijares-Vilarino** [Qiskit/qiskit-terra#8401]
For all classes in the [Circuit Library](https://qiskit.org/documentation/apidoc/circuit_library.html) the docstring from the `__init__` method is placed just after the `Class` one but they don't always mesh well together.

For example in the [XGate page](https://qiskit.org/documentation/stubs/qiskit.circuit.library.XGate.html#qiskit.circuit.library.XGate) you are given a full explanation of the gate (from `XGate Class` docstring) and then, just before the list of methods and out of nowhere, there is the message (from `__init__` docstring)

> Create new X gate.

All of this can be better seen in the following image:

![Image](https://user-images.githubusercontent.com/106545082/181004974-9f7e6da7-d66f-4485-ac56-4e581f1934e5.png)

**Comment by Cryoris** [comment]
Agreed this looks weird. In many places we already omit this first sentence in the docstring of the initializer to avoid this setting you show here.
